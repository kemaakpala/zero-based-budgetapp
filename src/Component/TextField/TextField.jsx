import PropTypes from "prop-types";

const TextField = ({label, inputName, placeholder, variant}) => (
  <>
    <label
      className={`${variant}-income-fields__label`}
      htmlFor={inputName}
    >
      {label}
    </label>
    <input
      className={`form-control ${variant}-income-fields__input`}
      id={inputName}
      type="text"
      name={inputName}
      placeholder={placeholder}
    />
  </>
);

TextField.propTypes = {
  label: PropTypes.string.isRequired,
  inputName: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  variant: PropTypes.string
};

TextField.defaultProps = {
  placeholder: "Please Enter Value Here",
  variant: "primary"
};

export default TextField;
