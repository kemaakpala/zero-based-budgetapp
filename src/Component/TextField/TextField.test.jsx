import { render } from "@testing-library/react";
import TextField from "./TextField";

describe("TextField", () => {
  it("test textfield renders", () => {
    const {
      container: { firstChild },
    } = render(<TextField label="test" inputName="testName" />);
    expect(firstChild).toMatchSnapshot();
  });
});
