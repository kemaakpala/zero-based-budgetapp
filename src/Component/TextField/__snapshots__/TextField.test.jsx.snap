// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField test textfield renders 1`] = `
<label
  class="primary-income-fields__label"
  for="testName"
>
  test
</label>
`;
